src/Command/GenerateObjectAclCommand.php

```diff
@@ -50,14 +51,20 @@ class GenerateObjectAclCommand extends QuestionableCommand
     private $aclObjectManipulators = [];
 
     /**
-     * @var RegistryInterface
+     * @var RegistryInterface|ManagerRegistry|null
      */
     private $registry;
 
-    public function __construct(Pool $pool, array $aclObjectManipulators, RegistryInterface $registry = null)
+    /**
+     * @param RegistryInterface|ManagerRegistry|null $registry
+     */
+    public function __construct(Pool $pool, array $aclObjectManipulators, $registry = null)
     {
         $this->pool = $pool;
         $this->aclObjectManipulators = $aclObjectManipulators;
+        if (null !== $registry && (!$registry instanceof RegistryInterface || !$registry instanceof ManagerRegistry)) {
```

Comment:
An `E_USER_DEPRECATED` error needs to be triggered if a `RegistryInterface` instance is passed.